code/game/machinery/chemHandC.dm

```diff
@@ -0,0 +1,295 @@
+		held_container = null
+		had_item = TRUE
+	toggle()
+	had_item = FALSE
+
+/obj/machinery/chemheater/verb/toggle()
+	set src in view(1)
+	set name = "Toggle heater"
+	set category = "Object"
+
+	if(!held_container && heating) //For when you take the beaker off but left the heater on
+		heating = !heating
+		overlays -= image(icon = icon, icon_state = "t[laser_kind]_beam")
+		processing_objects.Remove(src)
+		to_chat(usr, "<span class='notice'>You turn off \the [src].</span>")
+		update_icon()
+		return
+	else if(held_container)
+		heating = !heating
+		if(heating)
+			overlays += image(icon = icon, icon_state = "t[laser_kind]_beam")
+			processing_objects.Add(src)
+			to_chat(usr, "<span class='notice'>You turn on \the [src].</span>")
+		else
+			overlays -= image(icon = icon, icon_state = "t[laser_kind]_beam")
+			processing_objects.Remove(src)
+			to_chat(usr, "<span class='notice'>You turn off \the [src].</span>")
+		update_icon()
+		return
+	else
+		if(!had_item)
+			to_chat(usr, "<span class='notice'>\The [src] doesn't have anything to heat right now.</span>")
+
+/obj/machinery/chemheater/AltClick(mob/user)
+	if(!user.incapacitated() && Adjacent(user) && !(stat & (NOPOWER) && user.dexterity_check()))
+		toggle()
+		return
+	return ..()
+
+/*
+//Unused desired temp setting. Maybe useful in the future? Not likely since who doesn't want their coffee as hot as the sun?
+/obj/machinery/chemheater/verb/settemp(mob/user as mob)
+	set src in view(1)
+	set name = "Set temperature"
+	set category = "Object"
+
+	var/set_temp = input("Input desired temperature (20 to [TEMPERATURE_LASER] Celsius).", "Set Temperature") as num
+	if(set_temp>[TEMPERATURE_LASER] || set_temp<20)
+		to_chat(user, "<span class='notice'>Invalid temperature.</span>")
+		return
+	max_temperature = set_temp+273.15
+*/
+
+//Cooler
+
+/obj/machinery/chemcooler
+	name = "cryonic wave projector"
+	desc = "Ever want to see a microwave work in reverse? Well this machine is basically that. This machine could technically keep removing energy forever until it reaches absolute zero. Breaking physics and physicists since 2314 to current year and counting."
+	icon = 'icons/obj/chemHandC.dmi'
+	icon_state = "cooler"
+	icon_state_open = "cooler_open"
+	density = 1
+	anchored = 1
+	machine_flags = SCREWTOGGLE | CROWDESTROY | WRENCHMOVE | FIXED2WORK | EJECTNOTDEL
+	use_power = 1
+	idle_power_usage = 25
+	active_power_usage = 5000
+
+	var/max_temperature = 0 //You can make stuff REALLY cold
+	var/thermal_energy_transfer = -3000
+	var/scanner_kind = 0
+	var/onstage = null
+
+	var/obj/item/weapon/reagent_containers/held_container
+	var/cooling = FALSE
+	var/had_item = FALSE
+
+/obj/machinery/chemcooler/New()
+	. = ..()
+
+	component_parts = newlist(
+		/obj/item/weapon/circuitboard/chemcooler,
+		/obj/item/weapon/stock_parts/scanning_module,
+		/obj/item/weapon/stock_parts/capacitor
+	)
+	RefreshParts()
+	overlays += image(icon = icon, icon_state = "t[scanner_kind]_scanner") //Adds the side scanner
```

Comment:
This overlays += isn't really necessary considering you do the same thing in RefreshParts